src/mlpro/bf/math/unitconverter.py

```diff
@@ -0,0 +1,389 @@
+        type of the unit converter. Default: None.
+    p_unit_in : str
+        unit of the unit converter's input. Default: None.
+    p_unit_out : str
+        unit of the unit converter's output. Default: None.
+    p_logging
+        Log level (see constants of class Log). Default: Log.C_LOG_ALL.
+        
+    Attributes
+    ----------
+    C_TYPE : str
+        type of the base class. Default: 'UnitConverter'.
+    C_NAME : str
+        name of the unit converter. Default: ''.
+    C_UNIT_CONV_LENGTH : int
+        unit converter for length. Default: 0.
+    C_UNIT_CONV_PRESSURE : int
+        unit converter for pressure. Default: 1.
+    C_UNIT_CONV_CURRENT : int
+        unit converter for electric current. Default: 2.
+    C_UNIT_CONV_FORCE : int
+        unit converter for force. Default: 3.
+    C_UNIT_CONV_POWER : int
+        unit converter for power. Default: 4.
+    C_UNIT_CONV_MASS : int
+        unit converter for mass. Default: 5.
+    C_UNIT_CONV_TIME : int
+        unit converter for time. Default: 6.
+    C_UNIT_CONV_TEMPERATURE : int
+        unit converter for temperature. Default: 7.
+    """
+
+    C_TYPE                  = 'UnitConverter'
+    C_NAME                  = ''
+    C_UNIT_CONV_LENGTH      = 0
+    C_UNIT_CONV_PRESSURE    = 1
+    C_UNIT_CONV_CURRENT     = 2
+    C_UNIT_CONV_FORCE       = 3
+    C_UNIT_CONV_POWER       = 4
+    C_UNIT_CONV_MASS        = 5
+    C_UNIT_CONV_TIME        = 6
+    C_UNIT_CONV_TEMPERATURE = 7
+
+
+## -------------------------------------------------------------------------------------------------
+    def __init__(self,
+                 p_name:str,
+                 p_id:int=None,
+                 p_type:int=None,
+                 p_unit_in:str=None,
+                 p_unit_out:str=None,
+                 p_dt:float=0.01,
+                 p_logging=Log.C_LOG_ALL,
+                 **p_args) -> None:
+
+        super().__init__(p_name=p_name,
+                         p_id=p_id,
+                         p_type=p_type,
+                         p_unit_in=p_unit_in,
+                         p_unit_out=p_unit_out,
+                         p_dt=0,
+                         p_logging=Log.C_LOG_ALL,
+                         p_args=None)
+
+
+## -------------------------------------------------------------------------------------------------
+    def call(self, p_input, p_range=None):
+        """
+        This method provides a functionality to call the unit converter by giving an input value.
+
+        Parameters
+        ----------
+        p_input :
+            input value.
+        p_range :
+            not necessary for a unit converter. Default: None.
+
+        Returns
+        -------
+        output :
+            output value.
+        """
+        if self.get_type() == self.C_UNIT_CONV_TEMPERATURE:
+            output = self.temperature(p_input)
+        else:
+            output = self.scalar_conversion(p_input)
+        
+        return output
+
```

Comment:
Maybe if you want, you can use the default functionality of `__call__` So, instead of converter.call(input), you can just use directly the object name, converter(input).